service/controller/resource/certconfig/desired.go

```diff
@@ -37,26 +38,47 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interf
 	// Cluster deletion should not be affected only because some releases are
 	// missing or broken when fetching them from cluster-service.
 	if key.IsDeleted(&cr) {
-		r.logger.LogCtx(ctx, "level", "debug", "message", "not computing desired state of CertConfig CRs due to delete event")
+		r.logger.LogCtx(ctx, "level", "debug", "message", "not computing desired state", "reason", "the current state is used for deletion")
```

Comment:
I think `message` and `reason` are pretty much synonyms, but I'm ok going with this to see how it plays out. I wouldn't like to make `message` a label though, so not sure how useful the separation is instead of having word `because` in message instead.

Reply:
I started doing this in a couple of places where a single message started to become more complex. Examples are the change detection or the `tccpazs` resource handler. When looking at logs with `jq` or `gg` this was really useful and comfortable to digest while debugging.